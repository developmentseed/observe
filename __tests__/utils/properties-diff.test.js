/* global describe, expect, it */

import propertiesDiff from '../../app/utils/properties-diff'

describe('test properties diff', () => {
  it('should return a properties diff correctly', () => {
    const propsA = {
      'id': 'node/1',
      'version': 7,
      'building': 'yes',
      'addr:city': 'Seattle',
      'amenity': 'park'
    }
    const propsB = {
      'id': 'node/1',
      'version': 8,
      'building': 'residential',
      'addr:city': 'Seattle',
      'addr:state': 'WA'
    }
    const diff = propertiesDiff(propsA, propsB)
    expect(diff).toMatchSnapshot()
  })
})
