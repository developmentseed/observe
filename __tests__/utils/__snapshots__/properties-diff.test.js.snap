// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test properties diff should return a properties diff correctly 1`] = `
Object {
  "addr:city": Object {
    "unchanged": "Seattle",
  },
  "addr:state": Object {
    "added": "WA",
  },
  "amenity": Object {
    "deleted": "park",
  },
  "building": Object {
    "modifiedNew": "residential",
    "modifiedOld": "yes",
  },
}
`;
